lib/helpers/model/applyStaticHooks.js

```diff
@@ -1,11 +1,15 @@
 'use strict';
 
 const promiseOrCallback = require('../promiseOrCallback');
-const { queryMiddlewareFunctions, aggregateMiddlewareFunctions } = require('../../constants');
+const { queryMiddlewareFunctions, aggregateMiddlewareFunctions, modelMiddlewareFunctions, documentMiddlewareFunctions } = require('../../constants');
 
 const middlewareFunctions = [
-  ...queryMiddlewareFunctions,
-  ...aggregateMiddlewareFunctions
+  ...[
```

Comment:
You can just do `Array.from(new Set([...middleware1, ...middleware2]))` to avoid the need for `reduce()`.

Reply:
I also think your refactoring is a better approach! Thank you 👍